zooniverse/front-end-monorepo
# lib-user: Add stats custom date range with Calendar

## Package
- lib-user

## Describe your changes
- move MainContent components StyledButton and StyledTab to MainContent/components
- refactor MainContent helper functions
- add Calendar component to select custom date range
- refactor stories and specs as needed

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX?
- What user actions should my reviewer step through to review this PR?
  - go to user stats page or group stats page, select Custom option from date range select
  - select a custom date range
  - confirm stats update as expected
- Which storybook stories should be reviewed?
  - http://localhost:6008/?path=/story/components-shared-maincontent--default (somewhat, doesn't reflect custom date range selection on modal close)
- Are there plans for follow up PR’s to further fix this bug or develop this feature?
  - design / UI review:
    - Calendar modal Done button? - added in [4ffd3d0](https://github.com/zooniverse/front-end-monorepo/pull/6161/commits/4ffd3d0754f91ad2d5935c9591702d1115379018)
    - Select label with custom date range - refactored in [048d6df](https://github.com/zooniverse/front-end-monorepo/pull/6161/commits/048d6df1b03518af1a1fcf299ff0dc3f05944f2d)

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

## Refactoring
